web_monitoring/db.py

```diff
@@ -1,47 +1,608 @@
+    res = requests.get(url, auth=auth())
+    res.raise_for_status()
+    result = res.json()
+    data = result['data']
+    data['capture_time'] = parse_timestamp(data['capture_time'])
+    data['updated_at'] = parse_timestamp(data['updated_at'])
+    data['created_at'] = parse_timestamp(data['created_at'])
+    return result
+
+
+@ensure_credentials
+def post_version(page_id, uuid, capture_time, uri, version_hash,
+                 source_type, source_metadata=None):
+    """
+    Submit one new Version.
+
+    See :func:`post_versions` for a more efficient bulk importer.
+
+    Parameters
+    ----------
+    page_id : string
+        Page to which the Version is associated
+    uuid : string
+        a new, unique Version ID (UUID4)
+    start_date : datetime
+    end_date : datetime
+    uri : string
+        URI of content (such as an S3 bucket or InternetArchive URL)
+    version_hash : string
+        SHA256 of Version content
+    source_type : string
+        such as 'versionista' or 'internetarchive'
+    source_metadata : dict, optional
+        free-form metadata blob provided by source
+
+    Returns
+    -------
+    response : dict
+    """
+    # Do some type casting here as gentle error-checking.
+    version = {'uuid': str(uuid),
+               'capture_time': capture_time.isoformat() + 'Z',
+               'uri': str(uri),
+               'version_hash': str(version_hash),
+               'source_type': str(source_type),
+               'source_metadata': source_metadata}
+    url = f'{api_url()}/pages/{page_id}/versions'
+    res = requests.post(url, auth=auth(),
+                        headers={'Content-Type': 'application/json'},
+                        data=json.dumps(version))
+    res.raise_for_status()
+    return res.json()
 
 
+@ensure_credentials
 def post_versions(versions):
     """
-    Submit versions for importing into web-monitoring-db.
+    Submit versions in bulk for importing into web-monitoring-db.
 
     Parameters
     ----------
     versions : iterable
         iterable of dicts from :func:`format_version`
+
+    Returns
+    -------
+    response : dict
     """
     # Existing documentation of import API is in this PR:
     # https://github.com/edgi-govdata-archiving/web-monitoring-db/pull/32
-    url = WEB_MONITORING_CREATE_IMPORT_API.format(db_url=settings['db_url'])
-    return requests.post(url,
-                         auth=(settings['db_email'], settings['db_password']),
-                         headers={'Content-Type': 'application/x-json-stream'},
-                         data='\n'.join(map(json.dumps, versions)))
+    url = f'{api_url()}/imports'
+    res = requests.post(url, auth=auth(),
+                        headers={'Content-Type': 'application/x-json-stream'},
+                        data='\n'.join(map(json.dumps, versions)))
+    res.raise_for_status()
+    return res.json()
 
 
+@ensure_credentials
 def query_import_status(import_id):
```

Comment:
Seems like renaming to `get_import_status` would be more in-line with all the new functions.